### ISSUE TITLE
Pointer being freed was not allocated

### ISSUE THREAD

**Issue by dries-c** [netty/netty-incubator-codec-quic#652]
Running the following code on

```
netty: 4.1.89.Final 
codec version: 0.0.56.Final
arch: osx-aarch_64 (M1 Pro)
java sdk: openjdk 19.0.2
java runtime: openjdk 19.0.2
```

```java
QuicClientCodecBuilder codec = new QuicClientCodecBuilder()
        .maxIdleTimeout(2000, TimeUnit.MILLISECONDS)
        .initialMaxData(10000000)
        .initialMaxStreamDataBidirectionalLocal(1000000)
        .sslContext(QuicSslContextBuilder.forClient()
                .applicationProtocols("ng")
                .trustManager(InsecureTrustManagerFactory.INSTANCE)
                .build()
        );
```
or even
```java
QuicClientCodecBuilder codec = new QuicClientCodecBuilder()
```

Causes:
```shell
java(2728,0x289eaf000) malloc: *** error for object 0x9114004290000002: pointer being freed was not allocated
java(2728,0x289eaf000) malloc: *** set a breakpoint in malloc_error_break to debug
```

**Comment by normanmaurer** [comment]
Strange... this works for me. Do you have any other details ?

**Comment by normanmaurer** [comment]
maybe the full `hs_*` file ? Also can you reproduce this as well when you build this project by yourself ?

**Comment by dries-c** [comment]
Building the project itself seems to work just fine, all testcases pass.

**Comment by normanmaurer** [comment]
hmm ... I wonder if something did mess up while cross-compiling.

**Comment by normanmaurer** [comment]
I think this is fixed by 0.0.57.Final

**Comment by dries-c** [comment]
I tested it out on 0.0.57.Final, but it's still giving me the same error.
